snooty/parser.py

```diff
@@ -0,0 +1,362 @@
+import collections
+import docutils.nodes
+import logging
+import multiprocessing
+import os
+import pwd
+import subprocess
+from functools import partial
+from pathlib import Path, PurePath
+from typing import Any, Dict, Tuple, Optional, Set, List, Iterable
+from typing_extensions import Protocol
+import docutils.utils
+
+from . import gizaparser, rstparser, util
+from .gizaparser.nodes import GizaCategory
+from .types import Diagnostic, SerializableType, EmbeddedRstParser, Page, \
+    StaticAsset, ProjectConfigError, ProjectConfig
+
+RST_EXTENSIONS = {'.rst', '.txt'}
+logger = logging.getLogger(__name__)
+
+
+class JSONVisitor:
```

Comment:
Should this inherit from rstparser.Visitor?

Reply:
Nope; rstparser.Visitor is a Protocol for structural typing